File: c/cz/mg/vulkan/PFN_vkAllocateCommandBuffers.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkAllocateCommandBuffers_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pCommandBuffers, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkAllocateCommandBuffers f = (PFN_vkAllocateCommandBuffers)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkCommandBufferAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkCommandBuffer*)jniLongToPointer(pCommandBuffers))
    );
}

File: cz/mg/vulkan/PFNvkAllocateCommandBuffers.java
package cz.mg.vulkan;

public class PFNvkAllocateCommandBuffers extends VkFunctionPointer {
    public PFNvkAllocateCommandBuffers() {
    }

    public PFNvkAllocateCommandBuffers(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkAllocateCommandBuffers(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkAllocateCommandBuffers(long value) {
        setValue(value);
    }

    public PFNvkAllocateCommandBuffers(VkInstance instance) {
        super(instance, new VkString("vkAllocateCommandBuffers"));
    }

    public void call(VkDevice device, VkCommandBufferAllocateInfo pAllocateInfo, VkCommandBuffer pCommandBuffers, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pCommandBuffers != null ? pCommandBuffers.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getNullAddress());
    }

    private static native void call(long vkaddress, long device, long pAllocateInfo, long pCommandBuffers, long rval);
}
